crates/astria-sequencer/src/app/mod.rs

```diff
@@ -1301,12 +1297,15 @@ impl App {
                  just now or during the proposal phase",
             );
 
-        self.update_consensus_params_if_upgrade_due(
-            finalize_block.height,
-            &mut consensus_param_updates,
-        )
-        .await
-        .wrap_err("failed to update consensus params")?;
+        self.upgrades_handler
+            .finalize_block(
+                &mut self.state,
+                finalize_block.height,
+                &mut consensus_param_updates,
```

Comment:
`consensus_param_updates` will always be `None` here, as `PostTransactionExecutionResult` comes from `end_block` which always returns `None` for `consensus_param_updates` (which makes sense as we don't have any action types that update consensus params). change `upgrades_handler.finalize_block` to return the updated consensus params and remove the `&mut consensus_param_updates` arg. potentially change `upgrades_handler.finalize_block` to `upgrades_handler.end_block` and put the `consensus_param_updates` in `PostTransactionExecutionResult`, or just remove it from `PostTransactionExecutionResult`.

Reply:
Happy to do this.  I had assumed due to the `PostTransactionExecutionResult` already holding optional consensus param updates, that there were plans to allow actions to update these in the future.  If that's not the case, I'll make the changes you suggest but also move the `put_consensus_params` call down a level from `App::finalize_block` to `UpgradesHandler::end_block`, since if that's the only place where the params are updated, then it belongs in that method imo.